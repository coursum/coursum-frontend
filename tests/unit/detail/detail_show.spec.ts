import { shallowMount } from '@vue/test-utils';

import {
  classroomMock, curriculumCodeMock, registrationMock, relatedMock, tagMock, typesMock,
} from '../../data';
import { i18n } from '../../setup';

import DetailShow from '@/components/detail/detail_show.vue';

const options = {
  i18n,
  propsData: {
    registration: registrationMock,
    related: relatedMock,
    classroom: classroomMock,
    types: typesMock,
    tag: tagMock,
    curriculumCode: curriculumCodeMock,
  },
  mocks: {
    $t: (msg: string) => msg,
  },
};

describe('CourseShow', () => {
  it('should render correct markup when prop is correct', () => {
    const wrapper = shallowMount(DetailShow, options);

    expect(wrapper.element).toMatchSnapshot();
  });
});
