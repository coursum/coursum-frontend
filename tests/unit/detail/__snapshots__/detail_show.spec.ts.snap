// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseShow should render correct markup when prop is correct 1`] = `
<v-card-stub
  flat="true"
  loaderheight="4"
  tag="div"
>
  <table>
    <tr>
      <th>
        number
      </th>
       
      <td>
        <!---->
      </td>
    </tr>
     
    <tr>
      <th>
        suggestion
      </th>
       
      <td>
        <span>
          
            suggestion en
          
        </span>
      </td>
    </tr>
     
    <tr>
      <th>
        requirement
      </th>
       
      <td>
        <span>
          
            requirement en
          
        </span>
      </td>
    </tr>
     
    <tr>
      <th>
        requirement
      </th>
       
      <td>
        <!---->
      </td>
    </tr>
     
    <tr>
      <th>
        related
      </th>
       
      <td>
        <!---->
      </td>
    </tr>
     
    <tr>
      <th>
        classroom
      </th>
       
      <td>
        <span>
          
          classroom
        
        </span>
      </td>
    </tr>
     
    <tr>
      <th>
        types
      </th>
       
      <td>
        <!---->
      </td>
    </tr>
     
    <tr>
      <th>
        GIGA
      </th>
       
      <td>
        <span>
          
          true
        
        </span>
      </td>
    </tr>
     
    <tr>
      <th>
        curriculumCode
      </th>
       
      <td>
        <span>
          
          curriculumCode
        
        </span>
      </td>
    </tr>
  </table>
</v-card-stub>
`;
